internal/datanode/flow_graph_insert_buffer_node.go

```diff
@@ -384,8 +384,9 @@ func (ibNode *insertBufferNode) FillInSyncTasks(fgMsg *flowGraphMsg, seg2Upload
 	// the aim for taking all these actions is to guarantee that the memory consumed by delBuf will not exceed a limit
 	segmentsToFlush := ibNode.delBufferManager.ShouldFlushSegments()
 	for _, segID := range segmentsToFlush {
+		buf := ibNode.GetBuffer(segID)
 		syncTasks[segID] = &syncTask{
-			buffer:    nil, // nil is valid
+			buffer:    buf, // nil is valid
 			segmentID: segID,
 		}
 	}
```

Comment:
maybe merge `segmentsToFlush` and `syncSegmentIDs` below is a better choice to keep code simple and safe